apps/web/app/pages/sbtc-rewards/components/reward-value-displayer.tsx

```diff
@@ -2,17 +2,42 @@ import { ReactNode } from 'react';
 
 import { Flex, FlexProps, styled } from 'leather-styles/jsx';
 
+import { SkeletonLoader } from '@leather.io/ui';
+
 interface ValueDisplayerProps extends FlexProps {
   name: ReactNode;
   value: ReactNode;
+  isLoading?: boolean;
+  customSkeletonLayout?: ReactNode;
 }
-export function ValueDisplayer({ name, value }: ValueDisplayerProps) {
+export function ValueDisplayer({
+  name,
+  value,
+  isLoading = false,
+  customSkeletonLayout,
+}: ValueDisplayerProps) {
+  function ValueDisplayerContent() {
+    if (customSkeletonLayout) {
+      if (isLoading) {
+        return customSkeletonLayout;
+      }
+
+      return <styled.span textStyle="label.01">{value}</styled.span>;
+    }
+
+    return (
+      <SkeletonLoader height="15" width="80" isLoading={isLoading}>
+        <styled.span textStyle="label.01">{value}</styled.span>
+      </SkeletonLoader>
+    );
+  }
+
```

Comment:
Not supposed to embed components in components in react 🙅🏼

Can we make another component for this, rather than extending the base ValueDisplayer?

Reply:
fixed